File: quickstart-tutorials/lambda-api-mysql/src/lambdas/get-posts.ts
import { PrismaClient } from '@prisma/client';

// Instantiate the Prisma client.
const prisma = new PrismaClient();

const handler = async (event, context) => {
  try {
    // get all the posts data from the database using the Prisma client
    const posts = await prisma.post.findMany();

    return {
      statusCode: 200,
      headers: { 'Content-Type': 'application/json' },
      body: JSON.stringify({ message: 'success', data: posts })
    };
  } catch (error) {
    // If anything goes wrong, log the error.
    // You can later access the log data in the AWS console.
    console.error(error);
    return {
      headers: { 'Content-Type': 'application/json' },
      statusCode: 500,
      body: JSON.stringify({ message: 'error', error: error.message })
    };
  }
};

export default handler;

File: starter-projects/python-lambda-api/src/lambdas/get-posts.py
from pymongo import MongoClient
import os, json


client = MongoClient(os.environ['MONGODB_CONNECTION_STRING'])

def handler(event, context):

    try:
        db = client[os.environ['DATABASE_NAME']]
        posts = db.posts
        allPosts = posts.all()
        return {
            'statusCode': 200,
            'body': json.dumps(allPosts)
        }
    except Exception as e:
        return {
            'statusCode': 500,
            'body': json.dumps(str(e))
        }

if __name__ == "__main__":
    print(handler(None, None))
